src/main/java/io/temporal/internal/sync/SyncWorkflowContext.java

```diff
@@ -343,9 +343,8 @@ private ExecuteLocalActivityParameters constructExecuteLocalActivityParameters(
       activityTask.setInput(input.get());
     }
     RetryOptions retryOptions = options.getRetryOptions();
-    if (retryOptions != null) {
-      activityTask.setRetryPolicy(toRetryPolicy(retryOptions));
-    }
+    activityTask.setRetryPolicy(
+        toRetryPolicy(RetryOptions.newBuilder(retryOptions).validateBuildWithDefaults()));
```

Comment:
validateBuildWithDefaults needs to match same validation we do on server sides for activities?

Reply:
Not necessary as it is for local activity only. But I believe they do now.